src/components/flat-table/flat-table.style.js

```diff
@@ -7,6 +7,20 @@ import StyledFlatTableCheckbox from "./flat-table-checkbox/flat-table-checkbox.s
 import { baseTheme } from "../../style/themes";
 import { StyledFlatTableCell } from "./flat-table-cell/flat-table-cell.style";
 import cellSizes from "./cell-sizes.style";
+import Box from "../box";
+
+const StyledFlatTableBox = styled(Box)`
+  ${({ hasStickyFooter, theme }) =>
```

Comment:
I don't think this style should be hung off `hasStickyFooter` I think it would be fine as just

```
${({ theme }) =>
    css`
      box-shadow: inset 0px 0px 0px 1px ${theme.table.secondary};
      box-sizing: border-box;
    `}
 ```